src/renderer/src/routes/pages/Contacts/index.tsx

```diff
@@ -70,18 +73,39 @@ export const ContactsPage = () => {
     return groupContactsByFirstLetter
   }, [search, contacts])
 
-  const getNameInitials = (name: string) => {
+  const sortByBlockchain = useMemo(() => {
+    if (sortBlockchain === null) {
+      return (_a: TContactAddress, _b: TContactAddress) => 0
+    } else if (sortBlockchain) {
+      return (a: TContactAddress, b: TContactAddress) => a.address.localeCompare(b.address)
+    } else {
+      return (a: TContactAddress, b: TContactAddress) => b.address.localeCompare(a.address)
+    }
+  }, [sortBlockchain])
```

Comment:
Instead of returning a sorting function, it would be better to use this memo to actually do the sorting, because you don't need to sort for every re-render, as it stands now, React will run the sort function every time the component re-renders.

```suggestion
  const sortedSelectedContactAddresses = useMemo(() => {
    if (!selectedContact) return
  
    if (sortBlockchain === null) {
      return selectedContact.addresses
    }
    
    if (sortBlockchain) {
      return selectedContact.addresses.sort(a, b) => a.address.localeCompare(b.address)
    }
    
      return  selectedContact.addresses.sort(a, b) => b.address.localeCompare(a.address)
 
  }, [selectedContact, sortBlockchain])
```